### DISCUSSION TITLE
Asset volume permissions are confusing

### DISCUSSION THREAD

**Discussion by mmikkel** [craftcms/cms#9775] [Q&A]
Is it by design that the "Edit files uploaded by other users" user permission (which was added in 3.4.0) does not actually permit a user to edit an Asset unless the "Upload files" permission is also checked?

It's a bit confusing, since being able to upload new files doesn't _seem_ like it should be a prerequisite for being able to edit existing files' content. Also, other permissions – like "Edit images uploaded by other users" and "Replace files uploaded by other users" – does _not_ require the "Upload files" permission to work.

To be clear, the below permissions would *not* enable a user to edit existing files in a Volume (i.e. slideouts won't open on double-click and files will not be editable via their asset edit pages):

![CleanShot 2021-08-31 at 10 46 07](https://user-images.githubusercontent.com/298510/131472064-3637a114-8f92-4299-bf89-e9d7606443e2.png)

**Comment by brandonkelly** [comment]
Agree that is confusing. We should add a new explicit “Edit assets” permission, and restructure the asset permissions a bit, perhaps like this:

- View volume
  - View assets uploaded by other users
  - Edit assets
    - Edit assets uploaded by other users
  - Create subfolders
  - Remove files and folders
    - Remove files uploaded by other users
  - Upload files
  - Replace files
    - Replace files uploaded by other users
  - Edit images
    - Edit images uploaded by other users

**Reply by donkeycomm** [reply]
Would it be possible to add view/edit rights based on a craft website?
Different websites (languages) are sometimes used by different teams, and should not be able to see or edit each others content

**Reply by brandonkelly** [reply]
@donkeycomm Not directly, but you can do it by creating separate volumes that correspond to each site, and then limiting access to those volumes the same way you’re limiting access to sites.
